runtime/vm/createramclass.cpp

```diff
@@ -2405,7 +2405,7 @@ internalCreateRAMClassFromROMClassImpl(J9VMThread *vmThread, J9ClassLoader *clas
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
 			allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].prefixSize = 0;
 			allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].alignment = OMR_MAX(sizeof(J9Class *), sizeof(UDATA));
-			allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].alignedSize = sizeof(J9FlattenedClassCache) * (flattenedClassCache[0].offset > 0 ? flattenedClassCache[0].offset + 1 : 0);
+			allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].alignedSize = SIZEOF_FLATTENED_CLASS_CACHE_HEADER + sizeof(J9FlattenedClassCacheEntry) * (flattenedClassCache->numberOfEntries);
```

Comment:
This should be if the class is a valueType or if it contains flattened fields (ie `flattenedClassCache->numberOfEntries > 0`)

Reply:
Something like this? I am not sure which class I should determine if it's a value type.
`if (J9_IS_J9CLASS_VALUETYPE(flattenedClassCache->clazz) || (flattenedClassCache->numberOfEntries > 0)) {
				allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].prefixSize = 0;
				allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].alignment = OMR_MAX(sizeof(J9Class *), sizeof(UDATA));
				allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].alignedSize = SIZEOF_FLATTENED_CLASS_CACHE_HEADER + sizeof(J9FlattenedClassCacheEntry) * (flattenedClassCache->numberOfEntries);
				allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].address = NULL;
			}`

Reply:
@tajila By "valueROMClass" do you mean "romClass" since the first one is not declared or passed in as a parameter in this scope of internalCreateRAMClassFromROMClassImpl().

Reply:
yes